website/src/auth.tsx

```diff
@@ -135,7 +188,14 @@ export const UserProvider = (props: { children: any }) => {
       value={{
         user,
         setUser,
```

Comment:
Two things:
1. Should we do something here where we only expose the user to consumers if their email is verified?
2. Should we add a property to this hooks return like `unverifiedUser` that we set instead? That way pages that specifically need to handle users that have not verified their email yet still can do so.
3. Do we actually want to expose `setUser` to consumers? I would feel better if we knew that only this code here was allowed to `setUser` and only via the exposed operations.

Reply:
For (1) and (2), the AWS Cognito API alerts us to unverified users where it matters (most notably, for `GetUser` and `InitiateAuth`. Since our group and auth guards both require access to the JWT, they'll fail if we get a `UserNotConfirmedException`. Ive also added general handling for when we get back a `UserNotConfirmedException` that will alert the user and redirect to confirmation.

Implemented (3).